### DISCUSSION TITLE
Hot-reloading Configs

### DISCUSSION THREAD

**Discussion by stellarpower** [nushell/nushell#10904] [General]
IS it possible to reload nu's config files whilst the shell is running?

I tried this, which is similar to what I have in fish:

```nu
def reconf [] {
    source `~/.config/nushell/env.nu`
     source `~/.config/nushell/config.nu`
}
```
I get a stack overflow, which is not surprising given what I know about how the parser works.

IDK if there ought to be a FR to track infinite recursion in sourcing at all, but also would be keen to hear if anyone has a way to do it. Possibly spitting out to a secondary file I guess ,as thisi s how I loop over a oncfigs.d directory.

Thanks!

**Comment by fdncred** [comment]
Just add a keybinding like this one
```nushell
  {
    name: reload_config
    modifier: none
    keycode: f5
    mode: [ emacs vi_insert vi_normal ]
    event: [
      { edit: clear }
      { send: executehostcommand cmd: $"source ($nu.env-path); source ($nu.config-path)" }
    ]
  }
```

**Reply by stellarpower** [reply]
What editor is that for by the way? Would like something that works across different terminals, but this is nice if I can use it

**Reply by fdncred** [reply]
It has nothing to do with editors. It's a configuration keybinding in nushell's config.nu. It works cross-platform.

**Reply by stellarpower** [reply]
Right - I did not know those existed.

**Comment by fdncred** [comment]
We also have a recursion detector in nushell but it doesn't seem to work all the time.
